aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -700,16 +684,41 @@ module aptos_token::token {
         }
     }
 
+    public fun create_royalty(royalty_points_numerator: u64, royalty_points_denominator: u64, payee_address: address): Royalty {
+        assert!(royalty_points_denominator != 0, error::invalid_argument(EINVALID_ROYALTY_NUMERATOR_DENOMINATOR));
+        assert!(royalty_points_numerator <= royalty_points_denominator, error::invalid_argument(EINVALID_ROYALTY_NUMERATOR_DENOMINATOR));
+        assert!(account::exists_at(payee_address), error::invalid_argument(EROYALTY_PAYEE_ACCOUNT_DOES_NOT_EXIST));
```

Comment:
Is the royalty in the form of APT? If so, we need to ensure the payee account is registered to receive APT as well. You can use https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_account.move#L36

Reply:
ok good point! - adding that now

Reply:
Let's decouple coin type from token core contract. Token transaction can be executed in various type of coins so the royalty can be deducted in different coin types.